src/nvim/api/private/validate.h

```diff
@@ -0,0 +1,44 @@
+#ifndef NVIM_API_PRIVATE_VALIDATE_H
```

Comment:
introducing a validate.h module in anticipation that we will have more sophisticated validation things for C API in the future.